runtime/browser/runtime_download_manager_delegate.cc

```diff
@@ -178,12 +189,31 @@ void RuntimeDownloadManagerDelegate::ChooseDownloadPath(
 
   if (GetSaveFileName(&save_as))
     result = base::FilePath(std::wstring(save_as.lpstrFile));
-#else
-  NOTIMPLEMENTED();
-#endif
 
   callback.Run(result, content::DownloadItem::TARGET_DISPOSITION_PROMPT,
                content::DOWNLOAD_DANGER_TYPE_NOT_DANGEROUS, result);
+#elif defined(OS_LINUX) && !defined(OS_TIZEN)
+  content::WebContents* contents = item->GetWebContents();
```

Comment:
I dislike that we create a dialog here. 
Let's instead obtain 'Runtime' and invoke 'runtime->AddDownloadItem(item); // further UIDelegate is handling everything (including dialog creation)'